### PR TITLE
S05 us108 implement 30 days limit for ideas

### LINKED ISSUE #108: Ajouter une durée limite de 30 jours pour les idées, l'envoyer en archivage après les 30 jours

### PR DESCRIPTION
**Pull request by benrcd**

### CODE REVIEW

**Review by Spolito24 — APPROVED** [review]

**Review by AurelOconnell — APPROVED** [review]

File: src/Command/ArchiveIdeasCommand.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        $repository = $this->entityManager->getRepository(Idea::class);
+        $ideas = $repository->findAll();
+
+        foreach ($ideas as $idea) {
+            $endDate = $idea->getEndDate();
+
+            if ($endDate->format('Y-m-d') <= $currentDate->format('Y-m-d')) {
+                $idea->setArchived(true);
+            }
```
Comment (aldaitzwild) [inline src/Command/ArchiveIdeasCommand.php]: Do not compare formatted Date, just compare Date.
Cause when you compare two string : 
2023-5-2 > 2023-5-10

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Service/GetIdeaRemainingDays.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Idea;
+use DateTimeImmutable;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+class GetIdeaRemainingDays extends AbstractController
```
Comment (aldaitzwild) [inline src/Service/GetIdeaRemainingDays.php]: Why does a Service inherit from AbstractController ?

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Service/GetIdeaRemainingDays.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public function getRemainingDays(Idea $idea): int
+    {
+        $endDate = $idea->getEndDate();
+        $today = new DateTimeImmutable();
+        $interval = $today->diff($endDate);
+        $daysRemaining = $interval->days;
+
+        return $daysRemaining;
+    }
```
Comment (aldaitzwild) [inline src/Service/GetIdeaRemainingDays.php]: You should have this in your entity

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Entity/Idea.php [hunk 4]
```diff
@@ -164,4 +169,26 @@ public function removeSupporter(User $supporter): static
...
+    public function setEndDate(\DateTimeImmutable $endDate): static
+    {
+        $this->endDate = $endDate;
+
+        return $this;
+    }
+
+    public function getRemainingDays(Idea $idea): int
+    {
+        $endDate = $idea->getEndDate();
```
Comment (aldaitzwild) [inline src/Entity/Idea.php]: do you really need an idea here ? IN an idea ?? Couldn't $this wouldn't be enough ?

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Controller/IdeaController.php [hunk 5]
```diff
@@ -152,6 +155,7 @@ public function show(
             'idea' => $idea,
             'totalSupporters' => $totalSupporters,
             'isMember' => $isMember,
+            'daysRemaining' => $daysRemaining,
```
Comment (aldaitzwild) [inline src/Controller/IdeaController.php]: As you already pass the idea to twig, you don't need to pass your getRemainingDays, you can call it directly from twig

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Review by aldaitzwild — APPROVED** [review]

**Pull request merged**